src/deploy/D3MInit.sol

```diff
@@ -205,51 +217,35 @@ library D3MInit {
         dss.chainlog.setAddress(ScriptTools.stringToBytes32(string(abi.encodePacked(clPrefix, "_ORACLE"))), d3m.oracle);
     }
 
-    function initAave(
+    function initAavePool(
         DssInstance memory dss,
         D3MInstance memory d3m,
         D3MCommonConfig memory cfg,
-        D3MAaveConfig memory aaveCfg
+        D3MAavePoolConfig memory aaveCfg
     ) internal {
-        AavePlanLike plan = AavePlanLike(d3m.plan);
         AavePoolLike pool = AavePoolLike(d3m.pool);
-        ADaiLike adai = ADaiLike(aaveCfg.adai);
-
-        _init(dss, d3m, cfg, address(adai));
 
         // Sanity checks
```

Comment:
What sort of check do you want to do?

Reply:
At minimum we need to check that redeemable() returns a non empty address (it gives us an actual token)
If you want to go further we can also assert that name and symbol and decimals are not empty / no returned value.

Otherwise we are calling redeemable() without any form of sanity check to actually ensure that redeemable() 1) works and 2) gives us useful values.

That's how I would do it

Reply:
Calling the decimals and such functions implies it is non-zero address and at least looks somewhat like a token. I'm not sure what additional test should be performed.